src/SecureConnector.php

```diff
@@ -54,11 +56,30 @@ public function connect($uri)
             }
 
             // try to enable encryption
-            return $encryption->enable($connection)->then(null, function ($error) use ($connection) {
+            return $promise = $encryption->enable($connection)->then(null, function ($error) use ($connection, $uri) {
```

Comment:
Any reason to assign `$promise` here? Looks like a race condition bound to happen at line 73.

Reply:
This is intentional to either cancel the underlying TCP/IP connection attempt or the following TLS handshake. Depending on when ˋcancel()ˋ is called, it should internally ˋcancel()ˋ the underlying operation that is currently being executed.

Reply:
Makes perfect sense, didn't catch that 👍